agent/api/container_unix_test.go

```diff
@@ -0,0 +1,35 @@
+// +build !windows,!integration
+// Copyright 2014-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package api
+
+import (
+	"testing"
+
+	"github.com/aws/amazon-ecs-agent/agent/taskresource"
+	"github.com/aws/amazon-ecs-agent/agent/taskresource/cgroup"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestBuildResourceDependency(t *testing.T) {
```

Comment:
does `BuildResourceDependency` need to be tested by each resource we implement?

Reply:
That should not be the case since `BuildResourceDependency` has the same behavior for all types of resources.

Reply:
why not create a fake resource and make this platform independent?